src/Tools/dotnet-monitor/StartupHook/StartupHookValidator.cs

```diff
@@ -24,18 +24,26 @@ internal sealed class StartupHookValidator
 
 
         private readonly ILogger _logger;
+        private readonly IInProcessFeatures _inProcessFeatures;
         private readonly ISharedLibraryService _sharedLibraryService;
 
         public StartupHookValidator(
             ISharedLibraryService sharedLibraryService,
+            IInProcessFeatures inProcessFeatures,
             ILogger<StartupHookValidator> logger)
         {
             _logger = logger;
+            _inProcessFeatures = inProcessFeatures;
             _sharedLibraryService = sharedLibraryService;
         }
 
-        public async Task<bool> CheckAsync(IEndpointInfo endpointInfo, CancellationToken token)
+        public async Task<bool> CheckAsync(IEndpointInfo endpointInfo, CancellationToken token, bool logInstructions = false)
         {
+            if (!_inProcessFeatures.IsStartupHookRequired)
```

Comment:
I think I'd rather have the callers be responsible for this check rather than having them blindly invoke the `StartupHookValidator` and rely on its return value when startup hooks are not necessary. The return value indicates whether the startup hook was configured correctly or not. Returning false when startup hooks are not required tells the caller that startup hooks were not configured (but that wasn't really checked at all).